ecs-logs-collector.sh

```diff
@@ -27,7 +27,8 @@ export LC_ALL="C"
 
 # Common options
 curdir="$(dirname $0)"
-infodir="${curdir}/collect"
+instanceid="$(hostname -f)"
```

Comment:
"hostname -f" provides the host name of an instance, containing it's IP address and region. This is not it's instance ID. If you have 2 VPCs with the same IP address ranges, you could end up with 2 instances with the same host name.

In order to get the actual, unique, instance ID, you can query the instance metadata as follows.
curl http://169.254.169.254/latest/meta-data/instance-id
http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-instance-metadata.html

You can also find the instance ID in  /var/lib/cloud/data/instance-id on instances running cloud-init.